### DISCUSSION TITLE
SHA-3 hardware acceleration

### DISCUSSION THREAD

**Discussion by vk18anubhav** [openssl/openssl#24234] [Q&A] [answered]
Is hardware acceleration supported for SHA-3 in Openssl ?

**Comment by tom-cosgrove-arm** [comment, marked as answer]
There is hardware acceleration for SHA-3 for 64-bit Arm systems (AArch64) in `keccak1600-armv8.pl`. There are also versions for other CPUs in similarly-named files

**Reply by vk18anubhav** [reply]
Hi Tom,

Thanks for answering this. Where can I get information on configurations to enable these ?

Regards

**Reply by slontis** [reply]
Unless you are using the config option 'no-asm' it will use hardware acceleration if it detects the hardware capabilities automatically.
